### PR TITLE
Display empty previous data point on trend line when only one data point exists.

### PR DESCRIPTION
**Pull request by dsjen**
@rocha @clintonb @marcotuts 

FYI: @shnayder 

Example of engagement trend display with empty data point:
screen shot 2014-10-06 at 1 39 43 pm

Please note that the data table includes this empty data (not in CSV download).

### CODE REVIEW

File: analytics_dashboard/static/js/views/trends-view.js [hunk 1]
```diff
@@ -68,9 +68,6 @@ define(['bootstrap', 'd3', 'jquery', 'moment', 'nvd3', 'underscore', 'views/attr
                 // Remove the grid lines
                 canvas.selectAll('.nvd3 .nv-axis line').remove();
 
-                // Remove max value from the X-axis
-                canvas.select('.nvd3 .nv-axis.nv-x .nv-axisMaxMin:nth-child(3)').remove();
```
Comment (clintonb) [inline analytics_dashboard/static/js/views/trends-view.js:72]: Why is this being removed?

File: analytics_dashboard/courses/tests/utils.py [hunk 2]
```diff
@@ -42,6 +42,24 @@ def get_mock_enrollment_summary_and_trend(course_id):
     return get_mock_enrollment_summary(), get_mock_enrollment_data(course_id)
 
 
+def get_mock_enrollment_data_small(course_id):
+    single_enrollment = get_mock_enrollment_data(course_id)[-1]
+    empty_enrollment = {
```
Comment (clintonb) [inline analytics_dashboard/courses/tests/utils.py:47]: Does the API actually return a zero enrollment data point?

File: analytics_dashboard/static/js/views/trends-view.js (hunk 1 above)
Comment (dsjen) [inline analytics_dashboard/static/js/views/trends-view.js:72]: In the case that there were few points (e.g. two points), the chart looked very odd with only a start date as the last tick was removed.  By deleting this line, we get at both start and end tick marks.  I didn't see any display issues with larger datasets too.  Are there concerns?

File: analytics_dashboard/courses/tests/utils.py (hunk 2 above)
Comment (dsjen) [inline analytics_dashboard/courses/tests/utils.py:47]: No, the API doesn't.  This method provides the expected trend data, where the empty data point is inserted by the presenter in the case the API returns a single data point.

Comment (clintonb) [inline analytics_dashboard/courses/tests/utils.py:47]: If that's the case, please rename this function to make it clear that the returned value is data expected from the presenter, not the API method.

File: analytics_dashboard/static/js/views/trends-view.js (hunk 1 above)
Comment (clintonb) [inline analytics_dashboard/static/js/views/trends-view.js:72]: No concerns. It simply was not clear that you combined the axis clipping and minimal data tasks.

File: analytics_dashboard/courses/tests/utils.py (hunk 2 above)
Comment (dsjen) [inline analytics_dashboard/courses/tests/utils.py:47]: Makes sense--updated.

File: analytics_dashboard/courses/presenters.py [hunk 3]
```diff
@@ -52,24 +52,36 @@ def get_activity_types(self):
 
         return activities
 
+    def _build_trend_week(self, trend_types, week_ending, api_trend):
+        trend_week = {'weekEnding': week_ending.isoformat()}
```
Comment (rocha) [inline analytics_dashboard/courses/presenters.py:56]: Can you add a short comment about why we are doing this?

Comment (rocha) [inline analytics_dashboard/courses/presenters.py:56]: Nevermind, it is clear from the use below.

**Comment by rocha** [comment]
Looks good to me.

**Comment by dsjen** [comment]
@clintonb, do you have additional comments?

**Comment by clintonb** [comment]
Nope. :+1:

On Tuesday, October 7, 2014, Dennis Jen notifications@github.com wrote:

> @clintonb https://github.com/clintonb, do you have additional comments?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/edx/edx-analytics-dashboard/pull/157#issuecomment-58197555
> .

## 

Clinton Blackburn

**Pull request merged**